### PR TITLE
docs: Add SDD for 1609 feature request

### LINKED ISSUE #1609: Add apk name or module name to result matrix Result maxrix
**Author the user story for this feature**

As a [**Developer**], I want to [**see which module failed when there are lots of additional apks**] so I can [**understand which module has failed without opening result urls or reports**].

**Describe the solution**

```
┌─────────┬──────────────────────┬────────────────────────────┬───────────────────────────────────────────┐
│ OUTCOME │      MATRIX ID       │      TEST AXIS VALUE       │               TEST DETAILS                │
├─────────┼──────────────────────┼────────────────────────────┼───────────────────────────────────────────┤
│ failure │ matrix-b1fizzmw53qka │ NexusLowRes-29-de-portrait │ 82 test cases failed, 206 passed, 1 flaky │
│ failure │ matrix-wq8upcrn5uoca │ NexusLowRes-29-de-portrait │ 5 test cases failed, 4 passed             │
│ failure │ matrix-2n50loc2ze1h4 │ NexusLowRes-29-de-portrait │ 1 test cases failed, 6 passed             │
│ success │ matrix-3eo91na6dzmd1 │ NexusLowRes-29-de-portrait │ 8 test cases passed                       │
│ failure │ matrix-1ztkep4prcq43 │ NexusLowRes-29-de-portrait │ 1 test cases failed                       │
│ failure │ matrix-w9wpp0jwt4yba │ NexusLowRes-29-de-portrait │ 23 test cases failed, 6 passed            │
│ success │ matrix-30cqwb4sc1rqh │ NexusLowRes-29-de-portrait │ 17 test cases passed                      │
│ success │ matrix-32kzu6e7a1t3h │ NexusLowRes-29-de-portrait │ 2 test cases passed                       │
│ failure │ matrix-3dp8qqblpyav7 │ NexusLowRes-29-de-portrait │ 20 test cases failed, 12 passed           │
│ failure │ matrix-zw52wsouk4zma │ NexusLowRes-29-de-portrait │ 1 test cases failed                       │
│ INVALID │ matrix-36qxkc2szybc6 │                            │ NO_TEST_RUNNER_CLASS                      │
└─────────┴──────────────────────┴────────────────────────────┴───────────────────────────────────────────┘
```

From this table, it is hard to find out which matrix is reponsible for which additional apk 😄 

**Describe alternatives considered**

### PR DESCRIPTION
**Pull request by pawelpasterz**
Fixes #1609 

Adds SDD for [Add apk name or module name to result matrix Result maxrix](https://github.com/Flank/flank/issues/1609)

You can see it [HERE](https://github.com/Flank/flank/blob/1609-add-apk-name-to-result-table-sdd/docs/feature/1609-add-apk-name-to-result-matrix.md)

### CODE REVIEW

File: docs/feature/1609-add-apk-name-to-result-matrix.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# Add apk name or module name to the result matrix
+
+Currently, the result matrix displays matrices ids only. When there are lots of additional test apks it's hard to distinguish which apk failed. Having apk file name associated with results (matrices) will give a better overview and will allow a user to find a failing apk module without the need of jumping between result URLs.
```
Comment (Sloox) [inline docs/feature/1609-add-apk-name-to-result-matrix.md:3]:
```suggestion
Currently, the result matrix displays matrices ids only. When there are lots of additional test apks it's hard to distinguish which apk has failed. Having the apk file name associated with the results (matrices) will give a better overview and will allow a user to find a failing apk module without the need of jumping between result URLs.
```

File: docs/feature/1609-add-apk-name-to-result-matrix.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+└─────────┴──────────────────────┴────────────────────────────┴───────────────────────────────────────────┘
+```
+
+# References
+
+- [Issue #1609](https://github.com/Flank/flank/issues/1609)
+
+# Motivation
+
+Flank gives a possibility to run multiple apks in parallel within a single test run. This feature is extensively used by companies with big projects with lots of additional modules. Unfortunately, processing results in the current form is neither easy nor straightforward. The user needs to associate matrix with related apk/module which can be time-consuming and error-prone. To achieve a better result overview, flank should add apk name to the both result table and output JSON.
```
Comment (Sloox) [inline docs/feature/1609-add-apk-name-to-result-matrix.md:30]:
```suggestion
Flank gives the possibility to run multiple apks in parallel within a single test run. This feature is extensively used by companies with big projects that include a multitude of additional modules. Unfortunately, processing results in the current form is neither easy nor straightforward. The user needs to associate the matrix with the related apk/modules which can be time-consuming and error-prone. To achieve a better result overview, flank should add the apk name to both the result table and the output JSON.
```

File: docs/feature/1609-add-apk-name-to-result-matrix.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+# Goals
+
+Add an associated apk file name to the result table and output JSON.
+
+# Design
+
+`Flank` is not aware of any project, modules, etc so it can't find module names on its own, in a reasonable way, without breaking backward compatibility.
+
+On the other hand finding, the apk file name is simple and straightforward. To achieve it we need:
```
Comment (Sloox) [inline docs/feature/1609-add-apk-name-to-result-matrix.md:40]:
```suggestion
On the other hand, finding the apk file name is simple and straightforward. To achieve it we need:
```

File: docs/feature/1609-add-apk-name-to-result-matrix.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+1. add new field `app` to the `SavedMatrix` data class
+2. update private extension function `SavedMatrix#updateProperties` to extract apk file name from the `newMatrix`
+   * find app used during tests under the `TestMatrix` -> `TestSpecification`
+   * once we have `TestSpecyfication`, iterate over all specs (android/ios/instrumentation/etc.) and find non-null
```
Comment (Sloox) [inline docs/feature/1609-add-apk-name-to-result-matrix.md:44]:
```suggestion
   * once we have `TestSpecification`, iterate over all specs (android/ios/instrumentation/etc.) and find non-null
```

File: docs/feature/1609-add-apk-name-to-result-matrix.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+   * since we can run only one spec within a matrix, only one (spec) is non-null
+   * fetch `appApk` from spec (this is full `gs://` path) `TestSpecyfication#appApk`
```
Comment (Sloox) [inline docs/feature/1609-add-apk-name-to-result-matrix.md:46]:
```suggestion
   * fetch `appApk` from spec (this is full `gs://` path) `TestSpecification#appApk`
```
The y in the middle here is incorrect. If the class is called this we should change it :)

**Review by Sloox — CHANGES_REQUESTED** [review]

File: docs/feature/1609-add-apk-name-to-result-matrix.md (hunk 5 above)
Comment (pawelpasterz) [reply docs/feature/1609-add-apk-name-to-result-matrix.md:46]: Nah, that my mistake, thanks!

**Review by Sloox — APPROVED** [review]
LGTM 🚀

**Pull request merged**